common/frame.go

```diff
@@ -1889,8 +1891,7 @@ func (f *Frame) WaitForNavigation(opts goja.Value) *goja.Promise {
 				}
 			}
 		case <-timeoutCtx.Done():
-			handleTimeoutError(timeoutCtx.Err())
-			return nil, nil
+			return nil, handleTimeoutError(timeoutCtx.Err())
```

Comment:
Shouldn't it *not* return `nil` and a `nil` error if the timeout isn't DeadlineExceeded? 

no idea what this code does, just it seems .... uh ... strange.

Reply:
The first value is a `*Response`, which would be `nil` at this point anyways. If the timeout isn't `DeadlineExceeded`, it probably means the context was cancelled (e.g. iteration done or shutdown), so returning `nil, nil` also makes sense in that case. What do you think it should return instead?

We could return `resp, handleTimeoutError(timeoutCtx.Err())` on line 1904, though. The promise could still be rejected, but at least there's a chance they would have a response to work with. :man_shrugging:

Reply:
:shrug: 

Arguably you can return the error ... the code should not be executing at this point (*likely* there is still place for a race here). At worst the user will see a bunch of errors that are "context canceled".

But in the other case they could see errors as the response is `nil` but is still used for something. 

So I would argue them seeing the timeout errors will be better. 

In this case I would argue it will be nice to be able to check that k6 is stopping/has stopped and prevent the whole race condition to begin with :shrug:.

So I don't know - maybe just leave it as is and add a comment to be fixed at some point in the future

Reply:
In general it would be better to return an error. In the case it doesn't look like it really matters.

I think the problem is that we could easily forget that we're not returning an error which might be confusing in the future when we look back at this code. I vote to either return the original error if it's not `context.DeadlineExceeded` or log an error and return nil... but it's not idiomatic go to return a `nil` response and `nil` error... i think 🤷‍♂️

Reply:
> But in the other case they could see errors as the response is nil but is still used for something.

Yeah, that's a good point. I changed it in 65fbc97. Thanks!